clients/src/main/java/org/apache/kafka/common/utils/Bytes.java

```diff
@@ -141,7 +141,33 @@ private static String toString(final byte[] b, int off, int len) {
     }
 
     /**
-     * A byte array comparator based on lexicographic ordering.
+     * Increment the underlying byte array by adding 1.
+     *
+     * @param input - The byte array to increment
+     * @return A new copy of the incremented byte array.
+     */
+    public static Bytes increment(Bytes input) {
+        byte[] inputArr = input.get();
+        byte[] ret = new byte[inputArr.length + 1];
+        int carry = 1;
+        for (int i = inputArr.length - 1; i >= 0; i--) {
+            if (inputArr[i] == (byte) 0xFF && carry == 1) {
+                ret[i] = (byte) 0x00;
+            } else {
+                ret[i] = (byte) (inputArr[i] + carry);
+                carry = 0;
+            }
+        }
+        if (carry == 0) {
+            return wrap(Arrays.copyOf(ret, inputArr.length));
+        } else {
+            ret[0] = 1;
```

Comment:
I see what you're saying, but I am not sure it's proper to put the exception in the Bytes.increment as it's doing what it says it should do (ie: increment the byte array by 1). 

John and I noticed this before, but since CombinedKeySchema always starts with a positive integer, wrap-around won't affect us in either case. As John noticed 22 days ago:

```
Actually, I think this is not a problem. The key length is actually not arbitrary, since all keys must fit into byte arrays, and arrays in java cannot be larger than "max int", aka 0x7fff ffff Thus, we already have room to wrap around, with the first byte becoming 0x8f.
```

We're implicitly protected as long as the maximum of positive integers are always 0x7fff ffff.

Reply:
Good point. That seems to support an assertion / exception vs. adding behavior we don't expect to run though? If we overflow clearly there is a logic error somewhere in the callee and it would be better to find out directly vs. as a side effect of behavior we don't expect to exercise?

Reply:
Yep, fair enough. I'll make it such that it'll throw an error if it would result in a wrap around. The caller will have to allocate additional room if they want it to roll over.

Reply:
IndexArrayOutOfBounds? Or do you have something else in mind? I can't seem to find a reasonable choice from the list...

Reply:
I don't have a strong opinion. I suppose `IllegalArgumentException("overflow")` would be fine. That seems most appropriate, especially if we document that increment only works for inputs up to max int - 1.